Design Picker - Blank Canvas: Add tracks events

StefanNieuwenhuis opened the pull request:
### Changes proposed in this Pull Request

* use `isBlankCanvasDesign` util from `@automattic/design-picker` to determine if the blank canvas design is selected.
* update `trackingParams` object that's been fed to `calypso_newsite_complete` and `calypso_signup_complete` tracks events.
* add `isBlankCanvas` prop to `OnboardingCompleteParameters` interface.

### Testing instructions

#### A. `/new` flow

1. Open the Network tab in DevTools. Tick `Preserve log`.
1. Go through the `/new` flow.
1. When you reach the design picker page, select the blank canvas.
1. After the site creation, filter the network tab with `is_blank_canvas` and inspect the payload.
1. Verify that there are two events: `calypso_newsite_complete` and `calypso_signup_complete`, which both have 
   - `flow`: `gutenboarding`
   - `is_blank_canvas`: `true`.
1. Now, create a new site with anything but the blank canvas design.
1. After the site creation, filter the network tab with `is_blank_canvas` and inspect the payload.
1. Verify that there are two events: `calypso_newsite_complete` and `calypso_signup_complete`, which both have 
   - `flow`: `gutenboarding`
   - `is_blank_canvas`: `false`

#### B. `/start` flow

1. Open the Network tab in DevTools. Tick `Preserve log`.
1. Go through the `/start` flow. Select the build intent.
1. When you reach the design picker page, select the blank canvas.
1. After the site creation, filter the network tab with `is_blank_canvas` and inspect the payload.
1. Verify that there are an event: `calypso_signup_complete`, which has 
   - `flow`: `setup-site`
   - `is_blank_canvas`: `true`.
1. Now, create a new site with anything but the blank canvas design.
1. After the site creation, filter the network tab with `is_blank_canvas` and inspect the payload.
1. Verify that there are an event: `calypso_signup_complete`, which has 
   - `flow`: `setup-site`
   - `is_blank_canvas`: `false`


Related to #52289

StefanNieuwenhuis committed:
add is_blank_canvas to the tracking params

razvanpapadopol reviewed (dismissed):
The implementation looks fine for `/new`. I've also tested and steps 1-9 are all good.

However, for steps 10-11 to work (firing `calypso_signup_complete` with the blank canvas related prop on `/start`), you need to add the new prop to https://github.com/Automattic/wp-calypso/blob/trunk/client/lib/analytics/signup.js#L31-L78

Unrelated to this PR, when testing it, I've realised we're not recording the design selection correctly in `/new` so the value is always one step behind. Opened https://github.com/Automattic/wp-calypso/issues/52667 to track this.

StefanNieuwenhuis committed:
Add isBlankCanvas prop to /start

razvanpapadopol commented on client/signup/main.jsx:
```diff
@@ -428,6 +429,7 @@ class Signup extends React.Component {
 			isNewUser,
 			hasCartItems,
 			isNew7DUserSite,
+			isBlankCanvas: isBlankCanvasDesign( dependencies.selectedDesign ),
```
Shouldn't we do the check like in `client/landing/gutenboarding/hooks/use-on-site-creation.ts`? Or maybe update `isBlankCanvasDesign` to return `false` when called with `undefined`?
```suggestion
			isBlankCanvas: dependencies.selectedDesign && isBlankCanvasDesign( dependencies.selectedDesign ),
```

StefanNieuwenhuis replied:
> Or maybe update isBlankCanvasDesign to return false when called with undefined?

Thanks for the suggestion! I've updated the `isBlankCanvasDesign` util by making the design argument optional in 66decdb.

StefanNieuwenhuis committed:
Update isBlankCanvasDesign

We made the design argument optional and return false when none is provided. This enables us to remove verbose checks out when a design is undefined.

alshakero commented on client/landing/gutenboarding/hooks/use-on-site-creation.ts:
```diff
@@ -78,6 +79,7 @@ export default function useOnSiteCreation(): void {
 	const flowCompleteTrackingParams = {
 		isNewSite: !! newSite,
 		isNewUser: !! newUser,
+		isBlankCanvas: design && isBlankCanvasDesign( design ),
```
Since `isBlankCanvasDesign` accepts undefined.

```suggestion
		isBlankCanvas: isBlankCanvasDesign( design ),
```

StefanNieuwenhuis replied:
Thanks Omar! You're absolutely right and I committed your suggestion.

alshakero reviewed (commented):
Code LGTM! But I think it needs a rebase and newer testing steps.

StefanNieuwenhuis commented:
@alshakero, I rebased the PR, implemented your suggestions, and updated tested instructions. Can you give it another go?

alshakero reviewed (approved):
Reviewed and tested. LGTM!

roo2 commented:
@StefanNieuwenhuis team Ganon has two new members starting, we though that getting one of them to rebase this PR and take it across the line would be a good starting issue, any objections? https://github.com/Automattic/wp-calypso/issues/52289#issuecomment-1014116954

fushar commented:
Will take this up!

fushar commented:
I have rebased this PR and tested it. Could someone also test it as well? Thanks!
